src/Authentication/Authentication/custom/Find-MgGraphPermission.ps1

```diff
@@ -0,0 +1,226 @@
+# auto-completion works for commands such as Select-Object
+'Id', 'Consent', 'Name', 'Description', 'PermissionType' | foreach {
+    $typeParameters = @{
+        TypeName = 'Microsoft.Graph.Custom.Permission'
+        MemberType = 'NoteProperty'
+        MemberName = $_
+        Value = $null
+    }
+
+    Update-typedata @typeParameters -force
+}
+
+<#
+.SYNOPSIS
+Finds Microsoft Graph permissions based on search criteria.
+
+.DESCRIPTION
+For an application to access Microsoft Graph data, it must be granted consent for permissions to
+access that data by the user of the application or an administrator. Find-MgGraphPermission finds
+such permissions that match specified criteria. It returns information about each matching permission
+including the name of the permission, its unique identifier, and a description of the Microsoft Graph
+data access granted by that permission.
+
+The information returned by this command enables you to quickly identify the permission required
+for a script or application to access the particular types of data through Microsoft Graph. Once you've
+used the command to identify the permissions, you can use the names of those permission as a parameter to
+the Connect-MgGraph command to request the permission so that subsequent PowerShell commands or scripts
+may access the desired data from Microsoft Graph for instance. The unique identifier of a permission may
+also be used to configure consent for an application through additional PowerShell commands or other
+custom applications.
+
+Full details for all of the possible permissions and the access they authorize, including how to ensure
+that your application or script follows the best security pratice of using the least privilege necessary
+are found in the Microsoft Graph Permissions Reference documenation at https://docs.microsoft.com/en-us/graph/permissions-reference.
+
+In order to search for the specified permissions, Find-MgGraphPermission uses Microsoft Graph itself
+to obtain the latest complete set of permissions; if it is not able to access Microsoft Graph, the command
+then uses its most recent copy of the information as the source of data.
+
+Note that the results of Find-MgGraphPermission grouped by Delegated and Application permission types, and then sorted
+by permission name. They are not sorted by least-privileged access, so when multiple permissions appear to satisfy your
+script or application's requirements, consult the Microsoft Graph Permissions Reference documenation to identify the
+least-privileged permission for your use case.
+
+.PARAMETER SearchString
+The SearchString parameter allows you to specify a string such as 'user' or 'mail' that represents the subject or domain
+of the permission you're searching for. Since permissions usually have names such as 'User.Read' or 'Mail.ReadWrite', the
+command uses the SearchString parameter to return all permissions that contain the value specified for SearchString in the
+name of the permission.
```

Comment:
@FehintolaObafemi Clarification: I thought you said this searches the description field as well. I don't think that is the case, but just want to make sure.

Reply:
@TannerTrombley Adam and I opted to excluding the description field search.